infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/ServerExposer.java

```diff
@@ -178,6 +181,11 @@ public void expose(String namePrefix, Map<String, ? extends ServerConfig> server
     return exposedPorts;
   }
 
+  private boolean isTlsEnabled() {
+    String isTlsEnabled = System.getenv("CHE_INFRA_OPENSHIFT_ENABLE__TLS");
```

Comment:
You should inject value as named property instead of getting environment variable. I understand why you implement in this way =) So I'd move this logic to provision step, see [OpenShiftInfrastructureProvisioner](https://github.com/eclipse/che/blob/spi/infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInfrastructureProvisioner.java#L30). Then you wouldn't have any problem with property injection.

Reply:
thanks for the pointer. lemme try